qp_qiime2/qiime2.py

```diff
@@ -231,3 +233,71 @@ def pcoa(qclient, job_id, parameters, out_dir):
     ainfo = [ArtifactInfo('o-pcoa', 'ordination_results',
                           [(ffp, 'plain_text')])]
     return True, ainfo, ""
+
+
+def beta_correlation(qclient, job_id, parameters, out_dir):
+    """generate pcoa calculations
+
+    Parameters
+    ----------
+    qclient : qiita_client.QiitaClient
+        The Qiita server client
+    job_id : str
+        The job id
+    parameters : dict
+        The parameter values to rarefy
+    out_dir : str
+        The path to the job's output directory
+
+    Returns
+    -------
+    boolean, list, str
+        The results of the job
+    """
+    out_dir = join(out_dir, 'beta_correlation')
+    if not exists(out_dir):
+        mkdir(out_dir)
+
+    qclient.update_job_step(job_id, "Step 1 of 3: Collecting information")
+    artifact_id = parameters['i-distance-matrix']
+    artifact_info = qclient.get("/qiita_db/artifacts/%s/" % artifact_id)
+    dm_fp = artifact_info['files']['plain_text'][0]
+    dm_qza = join(out_dir, 'q2-distance.qza')
+    analysis_id = parameters['m-metadata-file']
```

Comment:
Remove this, in the artifact_info dict above, there should be the entry "analysis" which contains the analysis id.